src/guiFormSpecMenu.cpp

```diff
@@ -2065,48 +2065,30 @@ void GUIFormSpecMenu::regenerateGui(v2u32 screensize)
 }
 
 #ifdef __ANDROID__
-bool GUIFormSpecMenu::getAndroidUIInput()
+void GUIFormSpecMenu::handleAndroidInput(void *data, const std::string &text)
 {
-	/* no dialog shown */
-	if (m_JavaDialogFieldName == "") {
-		return false;
-	}
-
-	/* still waiting */
-	if (porting::getInputDialogState() == -1) {
-		return true;
-	}
-
-	std::string fieldname = m_JavaDialogFieldName;
-	m_JavaDialogFieldName = "";
-
-	/* no value abort dialog processing */
-	if (porting::getInputDialogState() != 0) {
-		return false;
-	}
+	GUIFormSpecMenu *menu = static_cast<GUIFormSpecMenu *>(data);
+	// No dialog shown
+	if (menu->m_JavaDialogFieldName.empty())
+		return;
 
-	for(std::vector<FieldSpec>::iterator iter =  m_fields.begin();
-			iter != m_fields.end(); ++iter) {
+	std::string fieldname = menu->m_JavaDialogFieldName;
+	menu->m_JavaDialogFieldName.clear();
 
-		if (iter->fname != fieldname) {
+	for (std::vector<FieldSpec>::iterator iter = menu->m_fields.begin();
+			iter != menu->m_fields.end(); ++iter) {
+		if (iter->fname != fieldname)
```

Comment:
would prefer an encompassing if statement here rather than a continue - less jumps makes code easier to understand imo